domain/domain.go

```diff
@@ -199,6 +199,12 @@ func (do *Domain) loadInfoSchema(startTS uint64) (infoschema.InfoSchema, bool, i
 	if err != nil {
 		return nil, false, 0, nil, err
 	}
+	schemaTs, err := m.GetTimestampForSchemaVersion(neededSchemaVersion)
```

Comment:
I think another method is that. we can just store the needed info, `startTS` or `commitTS` or whatever, into schemadiff. Check `GetSchemaVersionWithNonEmptyDiff()`, you are able to retrive the correct schema diff.